test/functional/JIT_Test/src/jit/test/jitserver/JITServerTest.java

```diff
@@ -133,10 +134,25 @@ private static String generatePortOption() {
 		// The only way to avoid most of the races is to have the server choose a random port (and retry if it is busy) and for us read the port
 		// from the server log. We still need to worry about tests where we stop and restart the server, because we need to keep using the same
 		// port but someone may grab it in the interim.
-		final int randomPort = EPHEMERAL_PORTS_START + new Random().nextInt(EPHEMERAL_PORTS_LAST - EPHEMERAL_PORTS_START + 1);
+		int randomPort = EPHEMERAL_PORTS_START + new Random().nextInt(EPHEMERAL_PORTS_LAST - EPHEMERAL_PORTS_START + 1);
+		while (!isPortOpen(randomPort)) {
+			logger.info("Port " + randomPort + " is busy");
+			randomPort = EPHEMERAL_PORTS_START + new Random().nextInt(EPHEMERAL_PORTS_LAST - EPHEMERAL_PORTS_START + 1);
+		}
 		return String.format(JITSERVER_PORT_OPTION_FORMAT_STRING, randomPort);
 	}
 
+	private static boolean isPortOpen(int port) {
+		Socket s = null;
+		try {
+			s = new Socket((String)null, port);
+			s.close();
+			return false;
```

Comment:
Shouldn't the condition be reversed? If you manage to open and close a socket, then you should return true, so that you exit the while loop above

Reply:
one sec, I think I got it wrong. I need to understand what Socket does. Does it open a port in the current process, or does it try to connect to a port already in use

Reply:
Yes, it was one technique I found to determine whether or not the socket is in use. I could add a comment explaining the strange condition direction.

Reply:
On Linux there is this `lsof -i: port` command that determines what process is using a port

Reply:
Is there a simple way to write platform-specific code in Java, if only to restrict the use of that command to Linux?

Reply:
I don't work that much in Java, but maybe `System.getProperty("os.name")` could tell whether the application runs on Linux.
JITServer is only running on Linux at the moment, so this test won't be running on other platforms.
The, to run the "lsof" program you can use the ProcessBuilder approach like in other place of this test

Reply:
The java preprocessor can do it, however IBM Java 8 shares the same class libraries across all platforms so it will never work there. Using "os.name" is the best bet, and this technique is already in use in other places.